### DISCUSSION TITLE
0051 XLS-51d: NFToken Escrows

### DISCUSSION THREAD

**Discussion by mvadari** [XRPLF/XRPL-Standards#146] [XLS Proposals]
## The earlier version can be found by looking at the edit history.

**Comment by xVet** [comment]
Great addition!!

**Reply by LetsEffingGo** [reply]
Could this open up the possibility of trustless P2P trading?

**Comment by ObiajuluM** [comment]
Nicee

**Comment by tequdev** [comment]
Very interesting proposal!
It’s better to clarify that NFTs in escrow cannot be burned.

**Reply by mvadari** [reply]
Good point, added 🙂

**Comment by ckeshava** [comment]
Very nice proposal! Ease of trade in NFTs should help the liquidity of the ecosystem.

Are you planning to transfer the ownership of NFTs into the Escrow? We need to prevent all modifications (transfer, deletions, etc) whilst the NFT is inside the Escrow.

Alternatively, we could perform ownership pre-requisite checks on Payment transaction types. This could vet that NFTs under Escrow are not being transferred. How can we implement this proposal?

**Reply by mvadari** [reply]
Yes, the ownership of the NFT moves to the escrow. It defeats the purpose of an escrow if anything can affect the NFT while it's in escrow.

**Comment by tequdev** [comment]
Not in the scope of rippled, but how are NFTs in escrow supposed to be handled in Clio's NFT methods(nfts_by_issuer, nft_history, nft_info)?

**Reply by mvadari** [reply]
They should probably still show up in those methods.

**Comment by mvadari** [comment]
One problem that I realized with this spec: you could use this to force someone to take an NFT without going through the standard accept flow. There is no way to accept an escrow right now.

This problem needs to be solved before this spec can be implemented.

Some possible answers:
* Add a mode to escrows that forces them to be accepted by the destination in order to be transferred, and alongside it a new transaction, `EscrowAccept`, that can be submitted at any point that the escrow exists (but before it is finished) in order for the NFT(s) to be transferred
* The `EscrowFinish` transaction must be submitted by the destination

**Reply by mvadari** [reply]
[DepositAuth](https://xrpl.org/docs/concepts/accounts/depositauth/) supports the destination finishing an escrow as equivalent to accepting it, so perhaps that flow makes the most sense.

**Comment by shortthefomo** [comment]
I won't put my feelings hating on as they sway to IOU's first.


But some over sites, for finish events on this spec.

EscrowBurn - burn NFT on finish.
EscrowCross - cross any open order on completion (obv take the best one).
EscrowCross - spec currency only to consider.

**Reply by mvadari** [reply]
I don't understand the need for burning an NFT on finish - why not a) finish the NFT then burn, or b) burn the NFT instead of sticking it in escrow? Ditto with selling it on finish.

**Reply by shortthefomo** [reply]
If it's conditional means the condition would need to have been met for the action to take place.

**Reply by mvadari** [reply]
I don't understand your comment. Can you elaborate?

[a comment by an automated account is not shown]

**Reply by mvadari** [reply]
Already merged 🙂
